subprojects/core/src/main/java/org/gradle/api/internal/plugins/ExtensionsStorage.java

```diff
@@ -132,7 +132,7 @@ public Object findByName(String name) {
     }
 
     private <T> ExtensionHolder<T> wrap(String name, TypeOf<T> publicType, T extension) {
-        if (isDeferredConfigurable(extension)) {
+        if (isDeferredConfigurable(extension) && !name.equals("publishing")) {
```

Comment:
In theory there could be another `publishing` extension annotated with `@DeferredConfigurable`. We could check for the fully-qualified class name as a String but I'm not sure it's worth it.

Reply:
Done